src/Files.App/Utils/Terminal/ConPTY/Native/ConsoleApi.cs

```diff
@@ -0,0 +1,44 @@
+using Microsoft.Win32.SafeHandles;
+using System.Runtime.InteropServices;
+
+namespace Files.App.Utils.Terminal.ConPTY
+{
+	/// <summary>
+	/// PInvoke signatures for win32 console api
+	/// </summary>
+	static class ConsoleApi
+	{
+		internal const int STD_OUTPUT_HANDLE = -11;
+		internal const uint ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004;
+		internal const uint DISABLE_NEWLINE_AUTO_RETURN = 0x0008;
+
+		[DllImport("kernel32.dll", EntryPoint = "AllocConsole", SetLastError = true, CharSet = CharSet.Auto, CallingConvention = CallingConvention.StdCall)]
+		internal static extern bool AllocConsole();
+
+		[DllImport("kernel32.dll", SetLastError = true)]
+		internal static extern nint GetConsoleWindow();
+
+		[DllImport("kernel32.dll", SetLastError = true)]
+		internal static extern SafeFileHandle GetStdHandle(int nStdHandle);
+
+		[DllImport("kernel32.dll", SetLastError = true)]
+		internal static extern bool SetConsoleMode(SafeFileHandle hConsoleHandle, uint mode);
+
+		[DllImport("kernel32.dll", SetLastError = true)]
+		internal static extern bool GetConsoleMode(SafeFileHandle handle, out uint mode);
```

Comment:
You can use CsWin32 for this guys, but not deal breaking in this PR I think.

Reply:
Switched to CsWin32 for most methods 👍